File: src/main/java/com/gitlab/daring/image/command/CommandUtils.java
package com.gitlab.daring.image.command;

import one.util.streamex.StreamEx;
import org.bytedeco.javacpp.opencv_core.Mat;

import java.util.function.Consumer;

import static com.gitlab.daring.image.util.ExtStringUtils.splitAndTrim;
import static java.util.Arrays.stream;

public class CommandUtils {

    public static Command newCommand(Consumer<Mat> c) {
        return new SimpleCommand(new String[0]).withFunc(c);
    }

    public static StreamEx<String> splitScript(String script) {
        return splitAndTrim(script, "\n").remove(l -> l.startsWith("//"))
            .flatMap(l -> splitAndTrim(l, ";")).remove(c -> c.startsWith("-"));
    }

    public static String[] parseArgs(String argStr) {
        return splitAndTrim(argStr, ",").toArray(String.class);
    }

    public static int[] parseIntParams(String... ps) {
        return stream(ps).mapToInt(Integer::parseInt).toArray();
    }

    private CommandUtils() {
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/CommandUtils.kt
package com.gitlab.daring.image.command

import com.gitlab.daring.image.util.ExtStringUtils.splitAndTrim
import org.bytedeco.javacpp.opencv_core.Mat

object CommandUtils {

    fun newEnvCommand(f: (CommandEnv) -> Unit): Command {
        return SimpleCommand(emptyArray()).apply { func = f }
    }

    fun newCommand(c: (Mat) -> Unit): Command {
        return SimpleCommand(emptyArray()).withFunc(c)
    }

    fun splitScript(script: String): List<String> {
        return splitAndTrim(script, "\n")
                .filterNot { it.startsWith("//") }
                .flatMap { splitAndTrim(it, ";") }
                .filterNot { it.startsWith("-") }
    }

    fun parseArgs(argStr: String): Array<String> {
        return splitAndTrim(argStr, ",").toTypedArray()
    }

    fun parseIntParams(ps: Array<String>) = ps.map(String::toInt)

}